pallets/erc20-xcm-bridge/src/lib.rs

```diff
@@ -73,11 +80,14 @@ pub mod pallet {
 					ref data,
 				}) = multilocation.interior().into_iter().next_back()
 				{
-					if let Ok(content) = core::str::from_utf8(&data[0..10]) {
-						if content == "gas_limit:" {
-							let mut bytes: [u8; 8] = Default::default();
-							bytes.copy_from_slice(&data[10..18]);
-							return u64::from_le_bytes(bytes);
+					// [u8; 10] for UTF8 "gas_limit:" + u64 for the limit
+					if data.len() >= size_of::<[u8; 10]>() + size_of::<u64>() {
```

Comment:
It's an extra cost of calculation for nothing. Can you apply @tmpolaczyk idea instead ?

IIRC, the idea is to force the `data` type at compile time with something like `let data: &[u8; 32] = &data;`.

It's enough to be sure that we'll notice if the type changes, because then we'll get a compilation error.
Of course, you need to add a comment to this line to specify that, in it fail to compile, we must ensure that the new type has at least 18 bytes.

Reply:
A better solution, but one that isn't a one line fix, is to define a new function:

```rust
fn read_prefixed_u64(data: &[u8; 32], prefix: &str) -> Option<u64> {

}
```

with the benefit that inside this function we can use `data.get(10..18)?`, and the indexing will return `None` on out of bounds.

Reply:
That's a good solution. But I think in that case it will never be out of bounds if the arg is `[u8; 32]`, right? It should first throw a compile error if data changes